vsock_proxy/src/dns.rs

```diff
@@ -18,35 +18,42 @@ pub fn resolve(addr: &str, ip_addr_type: IpAddrType) -> VsockProxyResult<Vec<Dns
 
     // DNS lookup
     // It results in a vector of IPs (V4 and V6)
-    let ips = lookup_host(&addr).map_err(|_| "DNS lookup failed!")?;
+    let resolver = Resolver::new(ResolverConfig::default(), ResolverOpts::default()).unwrap();
+
+    let rresults: Vec<DnsResolveResult> = resolver
+        .lookup_ip(addr)
+        .map_err(|error| format!("DNS lookup failed! {:?}", error))?
```

Comment:
This changes the format of the error message when the lookup fails. This makes `test_resolve_invalid_domain` unhappy as it matches on `DNS lookup failed!`. So the test probably needs an update to account for the extra info about the error type. I believe this is what makes the integration tests fail.

Reply:
Nice catch! Fixed in the new revision (1415a7d).